packages/screens/Mini/components/GradientBox.tsx

```diff
@@ -0,0 +1,65 @@
+import { LinearGradient } from "expo-linear-gradient";
+import { StyleProp, View, ViewStyle } from "react-native";
+
+import { primaryColor, secondaryColor } from "../../../utils/style/colors";
+
+export enum GradientDirectionEnum {
+  "topBottom",
+  "bottomTop",
+  "leftDiagnol",
+  "rightDiagnol",
+}
+
+type GradientBoxProps = {
+  size?: number;
+  radius?: number;
+  direction?: GradientDirectionEnum;
+  colors: string[];
+  style?: StyleProp<ViewStyle>;
+};
+
+export default function GradientBox({
+  size,
+  radius,
+  direction = GradientDirectionEnum.topBottom,
+  colors = [],
+  style,
+}: GradientBoxProps) {
+  const startEnd = gradientPositionGenerator(direction);
+
+  return (
+    <View
+      style={[
+        {
+          backgroundColor: primaryColor,
+          width: size ?? 48,
+          height: size ?? 48,
+          borderRadius: radius ?? 6,
+        },
+        style,
+      ]}
+    >
+      <LinearGradient
+        start={startEnd.start}
+        end={startEnd.end}
+        colors={colors.length ? colors : [primaryColor, secondaryColor]}
+        style={{ flex: 1, borderRadius: radius ?? 6 }}
+      />
+    </View>
+  );
+}
+
+function gradientPositionGenerator(direction: GradientDirectionEnum) {
+  const coordinates =
```

Comment:
Maybe add a comment to explain what it does ? Because 4 nested ternary conditions is complicated to follow